pkg/lib/dialogs.tsx

```diff
@@ -153,41 +158,28 @@ export const useDialogs = () => {
 };
 
 export const WithDialogs = ({ children } : { children: React.ReactNode }) => {
-    const is_open = useRef(false); // synchronous
-    const resolveRef = useRef<((x: unknown) => void) | null>(null);
-    const rejectRef = useRef<((x: unknown) => void) | null>(null);
     const [dialog, setDialog] = useState<React.ReactNode>(null);
+    const shown = useRef<"show" | "run" | "close">("close");
```

Comment:
typedef feels a bit nicer and clearer for the reviewer. The `typeof shown.current` feels like a hack